url49584c1d41_Arrem_GGS_tgz-pJ8-com_gamezgalaxy_test_console_MainJ8/src/com/gamezgalaxy/GGS/defaults/commands/Stop.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2012 GamezGalaxy.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the GNU Public License v3.0
+ * which accompanies this distribution, and is available at
+ * http://www.gnu.org/licenses/gpl.html
+ ******************************************************************************/
+package com.gamezgalaxy.GGS.defaults.commands;
+
+import com.gamezgalaxy.GGS.API.CommandExecutor;
+import com.gamezgalaxy.GGS.API.plugin.Command;
+import java.io.IOException;
+
+public class Stop extends Command
+{
+	@Override
+	public String[] getShortcuts()
+	{
+		return new String[0];
+	}
+
+	@Override
+	public String getName()
+	{
+		return "stop";
+	}
+
+	@Override
+	public boolean isOpCommand()
+	{
+		return true;
+	}
+
+	@Override
+	public int getDefaultPermissionLevel()
+	{
+		return 100;
+	}
+
+	@Override
+	public void execute(CommandExecutor player, String[] args)
+	{
+		try {
+			player.getServer().Stop();
+		} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)